Sources/FoundationEssentials/Subprocess/Platforms/Subprocess+Unix.swift

```diff
@@ -0,0 +1,286 @@
+
+        public static var interrupt: Self { .init(rawValue: SIGINT) }
+        public static var terminate: Self { .init(rawValue: SIGTERM) }
+        public static var suspend: Self { .init(rawValue: SIGSTOP) }
+        public static var resume: Self { .init(rawValue: SIGCONT) }
+        public static var kill: Self { .init(rawValue: SIGKILL) }
+        public static var terminalClosed: Self { .init(rawValue: SIGHUP) }
+        public static var quit: Self { .init(rawValue: SIGQUIT) }
+        public static var userDefinedOne: Self { .init(rawValue: SIGUSR1) }
+        public static var userDefinedTwo: Self { .init(rawValue: SIGUSR2) }
+        public static var alarm: Self { .init(rawValue: SIGALRM) }
+        public static var windowSizeChange: Self { .init(rawValue: SIGWINCH) }
+    }
+
+    public func sendSignal(_ signal: Signal, toProcessGroup shouldSendToProcessGroup: Bool) throws {
+        let pid = shouldSendToProcessGroup ? -(self.processIdentifier.value) : self.processIdentifier.value
+        guard kill(pid, signal.rawValue) == 0 else {
+            throw POSIXError(.init(rawValue: errno)!)
+        }
+    }
+}
+
+// MARK: Environment Resolution
+extension Subprocess.Environment {
+    internal static let pathEnvironmentVariableName = "PATH"
+
+    internal func pathValue() -> String? {
+        switch self.config {
+        case .inherit(let overrides):
+            // If PATH value exists in overrides, use it
+            if let value = overrides[.string(Self.pathEnvironmentVariableName)] {
+                return value.stringValue
+            }
+            // Fall back to current process
+            return ProcessInfo.processInfo.environment[Self.pathEnvironmentVariableName]
+        case .custom(let fullEnvironment):
+            if let value = fullEnvironment[.string(Self.pathEnvironmentVariableName)] {
+                return value.stringValue
+            }
+            return nil
+        }
+    }
+
+    // This method follows the standard "create" rule: `env` needs to be
+    // manually deallocated
+    internal func createEnv() -> [UnsafeMutablePointer<CChar>?] {
+        func createFullCString(
+            fromKey keyContainer: Subprocess.StringOrRawBytes,
+            value valueContainer: Subprocess.StringOrRawBytes
+        ) -> UnsafeMutablePointer<CChar> {
+            let rawByteKey: UnsafeMutablePointer<CChar> = keyContainer.createRawBytes()
+            let rawByteValue: UnsafeMutablePointer<CChar> = valueContainer.createRawBytes()
+            defer {
+                rawByteKey.deallocate()
+                rawByteValue.deallocate()
+            }
+            /// length = `key` + `=` + `value` + `\null`
+            let totalLength = keyContainer.count + 1 + valueContainer.count + 1
+            let fullString: UnsafeMutablePointer<CChar> = .allocate(capacity: totalLength)
+            #if canImport(Darwin)
+            _ = snprintf(ptr: fullString, totalLength, "%s=%s", rawByteKey, rawByteValue)
```

Comment:
I wonder if there's any way to do this in a safer manner that doesn't rely on string formatting (which has a lot of pitfalls / sneaky issues in Swift)